### ISSUE TITLE
Changing the MapConfig from Hazelcast Mancenter leads to incorrect behavior

### ISSUE THREAD

**Issue by nikbali** [hazelcast/hazelcast#18123]
My environment configuration:
- Hazelcast Version = **3.12.5**
- Hazelcast Mancenter Version = 3.12.7
- 1 member cluster
- For connect to Hazelcast cluster Iam using - `HazelcastClient.newHazelcastClient(config)`

**To Reproduce**

I configure IMap in my Spring Boot application:

```
@Slf4j
@Configuration
public class HazelcastConfiguration {

    @Autowired
    private HazelcastInstance hazelcastInstance;

    @Bean
    @Qualifier(TEST_ADD_CONFIG)
    public IMap<String, SozProductsResponseDto> cacheBean() {

        final MapConfig mapConfig = new MapConfig(TEST_ADD_CONFIG);
        mapConfig.setEvictionPolicy(EvictionPolicy.LRU);
        mapConfig.setTimeToLiveSeconds(5);
        var maxSizeConfig = new MaxSizeConfig(20, USED_HEAP_PERCENTAGE);
        mapConfig.setMaxSizeConfig(maxSizeConfig);

        hazelcastInstance.getConfig().addMapConfig(mapConfig);
        return hazelcastInstance.getMap(TEST_ADD_CONFIG);
    }
}


```

I set up evict LRU policy by 20% percents of JVM heap size (`new MaxSizeConfig (20, USED_HEAP_PERCENTAGE)`)

I ran test and added over 100,000 elements to the map.
Everything was great. Eviction worked as it should be.

Then I updated MapConfig from Hazelcast Mancenter with new TTL value. (Changed to TTL=10)

<img width="1027" alt="Снимок экрана 2021-01-29 в 20 57 33" src="https://user-images.githubusercontent.com/24268582/106310449-9fb37a00-6274-11eb-933e-8048fa760534.png">

And when I tried to add 100000 elements to the map, I couldn't do that. There was only **271 elements.**

<img width="1768" alt="Снимок экрана 2021-01-29 в 21 00 46" src="https://user-images.githubusercontent.com/24268582/106310740-1486b400-6275-11eb-92b7-b160f77cc4ba.png">

I read ([stackoverflow question](https://stackoverflow.com/questions/52630083/problem-with-hazelcast-eviction-the-firsts-entries-and-small-size-in-configurati)) that this problem happens because Hazelcast splits each map into 271 partitions by default.
And for maps with max_size less than 271 `put` operation will not work
The reason is that all 271 partitions will be placed on 1 member.

**Problem**
But I didn't set a maximum limit of the map size. I set only one limit depending on heap size. (`new MaxSizeConfig (20, USED_HEAP_PERCENTAGE)`)

I think that when I have changed TTL value in Hazelcast Mancenter
It also affected a configuration property with maximum map size map limit.

Please, help me to understand this problem and find a solution.

**Comment by mmedenjak** [comment]
Hi @nikbali ! 

I think the problem is that the "update map config" feature promises more than it can deliver. Specifically, it says it will update the whole map configuration but actually some parts work and some don't. More concretely, I think changing the TTL value is not supported. @ahmetmircik can you confirm?

**Comment by ahmetmircik** [comment]
@mmedenjak I can't see a field for max-size-policy in the screenshot above, possibly it is sending a default value, may be it is PER_NODE since it is the first in enum, to MC and this changing the previously set max-size-policy and causing the issue. I think, this should be working in later versions of management center. This very case in the issue should work without any problem.

**Comment by mmedenjak** [comment]
Thanks, maybe @emre-aydin can comment with some thoughts and ideas.

**Comment by emre-aydin** [comment]
@ahmetmircik you're right. MC always sends the config with `"PER_NODE"` max size policy and max size=`2147483647`. I'm not sure why that is the case. Since we don't allow changing the max size policy and size via MC, I think it makes sense on cluster side to discard the setting coming from MC and not send them in next MC 3.12.z version. If it makes sense to you as well, I can send PRs to both MC and IMDG.

**Comment by ahmetmircik** [comment]
@emre-aydin Server side supports max size policy changes, for instance one can make a non-evictable IMap evictable at runtime. To me, if it is not a used feature, we can disable it, but i don't have supporting evidence for this.

**Comment by emre-aydin** [comment]
@ahmetmircik since the server side supports it, it makes sense to add the option to the UI. I will create an issue for it in MC repository so that we can fix it in the next version of MC.

**Comment by nikbali** [comment]
> @ahmetmircik you're right. MC always sends the config with `"PER_NODE"` max size policy and max size=`2147483647`. I'm not sure why that is the case. Since we don't allow changing the max size policy and size via MC, I think it makes sense on cluster side to discard the setting coming from MC and not send them in next MC 3.12.z version. If it makes sense to you as well, I can send PRs to both MC and IMDG.

I think that overridden my previously set configuration implicitly.
But why can't I add more than 271 entries to the map?
If you say that MC always sends configuration with "PER_NODE" policy of maximum size and maximum size = 2147483647

**Comment by promanenko** [comment]
Fixed in Management Center 4.2021.04 release
https://docs.hazelcast.org/docs/management-center/rn/index.html#4.2021.04
